pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -695,6 +695,12 @@ private void serializeAndSendMessage(MessageImpl<?> msg,
                 op = OpSendMsg.create(msg, null, sequenceId, callback);
                 final MessageMetadata finalMsgMetadata = msgMetadata;
                 op.rePopulate = () -> {
+                    if (msgMetadata.hasChunkId()) {
+                        // The message metadata is shared between all chunks in a large message
+                        // We need to reset the chunk id for each call of this method
+                        // It's safe to do that because there is only 1 thread to manipulate this message metadata
+                        finalMsgMetadata.setChunkId(chunkId);
```

Comment:
- Since multiple chunks use the same message meta, the chunk ID must be reset when resending messages, right?<sup>[1]</sup>
- So it's solving a different problem, right?<sup>[2]</sup>